### PR TITLE
Expansion of shorthands

### PR DESCRIPTION
**Pull request by ewilligers**
We check the values assigned to longhand properties
by the following shorthand properties:
- flex
- flex-flow
- inset
- list-style
- margin
- outline
- padding
- text-decoration

### CODE REVIEW

File: css/css-lists/parsing/list-style-shorthand.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<meta name="assert" content="list-style supports the full grammar '<'list-style-position'> || <'list-style-image'> || <'list-style-type'>'.">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/css/support/shorthand-testcommon.js"></script>
+</head>
+<body>
+<script>
+test_shorthand_value('list-style', 'square url("https://example.com/") inside', {
+  'list-style-position': 'inside',
+  'list-style-image': 'url("https://example.com/")',
```
Comment (emilio) [inline css/css-lists/parsing/list-style-shorthand.html]: Mind using a non-remote URL? Maybe a data URI?

I might be being over-paranoid, but I've had to change some of these tests so they don't crash on Firefox's automation (we crash as soon as Firefox tries to hit the network on automation).

**Review by emilio — APPROVED** [review]
Thanks as always Eric, these are awesome!

One suggestion, as I saw that you added `test_shorthand_value` in 6e2637cce43, very recently. Can we update it to:

 * Unset the property at the end of the tests, to prevent bugs.
 * Assert that there are no other properties that the shorthand expands to (that is, remove the longhands one by one, assert that the remaining declaration is empty).

I think this would make the tests even better, wdyt?

**Comment by emilio** [comment]
Maybe worth doing that in a different PR in any case, your call

File: css/css-lists/parsing/list-style-shorthand.html (hunk 1 above)
Comment (ewilligers) [reply css/css-lists/parsing/list-style-shorthand.html]: Changed to use {{host}}

**Comment by ewilligers** [comment]
> Assert that there are no other properties that the shorthand expands to (that is, remove the longhands one by one, assert that the remaining declaration is empty).

When I tried clearing any one longhand, it appeared to clear the shorthand.

**Comment by ewilligers** [comment]
> Unset the property at the end of the tests, to prevent bugs.

Done

**Comment by emilio** [comment]
> When I tried clearing any one longhand, it appeared to clear the shorthand.

Can you clarify? It totally does not. The shorthand will serialize as empty, but there should be declarations there (as in, `cssText` shouldn't be empty, `.length` should be != 0):

```
document.body.style = "";
document.body.style.length // 0
document.body.style.margin = "0px"
document.body.style.length // 4
document.body.style.marginTop = ""
document.body.style.length // 3
```

**Comment by ewilligers** [comment]
> Can you clarify? It totally does not. The shorthand will serialize as empty, ...

That's what I was trying.

Now added a check for `style.length` leaks in https://github.com/web-platform-tests/wpt/pull/19195

**Pull request merged**